internal/resultstore/resultstore.go

```diff
@@ -56,6 +60,58 @@ func (rs *ResultStore) generatePath(p Pkg) string {
 	return path
 }
 
+func (rs *ResultStore) SaveWriteBufferZip(ctx context.Context, p Pkg, fileName string, writeBufferPaths []string) error {
```

Comment:
A bit of a nit: may I suggest calling this something like: `SaveFilesToZip` or something like that?

Reply:
I called this SaveTempFilesToZip since this function shouldn't be reused and only works if the file is saved in the correct temp directory